app/components/post_body/system_message_helpers.js

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {Posts} from 'mattermost-redux/constants';
+import Markdown from 'app/components/markdown';
+
+const renderUsername = (value) => {
+    return (value[0] === '@') ? value : `@${value}`;
+};
+
+const formattedMessage = (localeHolder, elements, intl) => {
+    const {formatMessage} = intl;
+
+    return formatMessage(localeHolder, elements);
+};
+
+const renderHeaderChangeMessage = (postBodyProps, styles, intl) => {
+    const {onPress, postProps} = postBodyProps;
+    const {messageStyle, textStyles} = styles;
+
+    if (!postProps.username) {
+        return null;
+    }
+
+    const username = renderUsername(postProps.username);
+    const oldHeader = postProps.old_header;
+    const newHeader = postProps.new_header;
+    let localeHolder;
+
+    if (postProps.new_header) {
+        if (postProps.old_header) {
+            localeHolder = {
+                id: 'api.channel.post_update_channel_header_message_and_forget.updated_from',
+                defaultMessage: '{username} updated the channel header from: {oldHeader} to: {newHeader}',
+            };
+
+            return (
+                <Markdown
+                    baseTextStyle={messageStyle}
+                    onPostPress={onPress}
+                    textStyles={textStyles}
+                    value={formattedMessage(localeHolder, {username, oldHeader, newHeader}, intl)}
+                />
+            );
+        }
+
+        localeHolder = {
+            id: 'api.channel.post_update_channel_header_message_and_forget.updated_to',
+            defaultMessage: '{username} updated the channel header to: {newHeader}',
+        };
+
+        return (
+            <Markdown
+                baseTextStyle={messageStyle}
+                onPostPress={onPress}
+                textStyles={textStyles}
+                value={formattedMessage(localeHolder, {username, oldHeader, newHeader}, intl)}
+            />
+        );
+    } else if (postProps.old_header) {
+        localeHolder = {
+            id: 'api.channel.post_update_channel_header_message_and_forget.removed',
+            defaultMessage: '{username} removed the channel header (was: {oldHeader})',
+        };
+
+        return (
+            <Markdown
```

Comment:
instead of returning a `Markdown` component in every if statement, you could do something like this to clean it up a bit.

```js
let value;
if (...) {
   value = intl.formatMessage({
        id: t('the_string_id'),
        defaultMessage: 'the message associate with the string id',
        {var1, var2, var3, ...}
    );
}

if (value) {
    return (
        <Markdown
          ...
          value={value}
         />
    );
}

return null;
}

the the function `formattedMessage` is not even needed

Reply:
See my other comment for another option

Reply:
Done!